API Poll Participation

HankG opened the pull request:
There was no method in the original API spec to do this so this is what I came up with...

HankG committed:
Add Poll Voting Method to Interactions Endpoint

frankrousseau commented on spec/integration/api/post_interactions_controller_spec.rb:
```diff
@@ -291,4 +291,91 @@
+          access_token:   access_token
+        }
+      )
+      expect(response.status).to eq(422)
```
When url is wrong it seems better to send a 404 error.

HankG replied:
I've corrected this in the new version.

HankG commented:
I added a little extra handling around the find-post line to catch that "record not found" specifically so now it returns 404 in that instance.  I've also squashed this so it's ready for integration.

frankrousseau commented:
Great, thank you!
